src/apps/the-exchange-app/components/EnterAmount/EnterAmount.tsx

```diff
@@ -0,0 +1,100 @@
+import BodySmall from '../Typography/BodySmall';
+import SendArrow from '../../images/send-arrow.png';
+import ReceiveArrow from '../../images/receive-arrow.png';
+import { useContext, useState, useEffect } from 'react';
+import { SwapReceive } from '../../utils/types';
+import { SwapDataContext } from '../../context/SwapDataProvider';
+import { useEtherspotPrices } from '@etherspot/transaction-kit';
+import useOffer from '../../hooks/useOffer';
+
+type EnterAmountProps = {
+    type: SwapReceive;
+    tokenSymbol?: string;
+}
+
+const EnterAmount = ({ type, tokenSymbol }: EnterAmountProps) => {
+    const [isHover, setIsHover] = useState<boolean>(false);
+    const [isFocus, setIsFocus] = useState<boolean>(false);
+    const { amountSwap, setAmountSwap, amountReceive, setAmountReceive, swapToken, receiveToken, setBestOffer } = useContext(SwapDataContext);
+    const [inputValue, setInputValue] = useState<string>('0.00');
+    const { getPrice } = useEtherspotPrices();
+    const { getBestOffer } = useOffer(swapToken?.chainId || 0);
+
+    const getOffer = async () => {
+        const params = {
+            fromAmount: amountSwap?.tokenAmount || 0,
+            fromTokenAddress: swapToken?.address || '',
+            fromChainId: swapToken?.chainId || 0,
+            fromTokenDecimals: swapToken?.decimals || 0,
+            toTokenAddress: receiveToken?.address || '',
+            toChainId: receiveToken?.chainId || 0,
+            toTokenDecimals: receiveToken?.decimals || 0,
+        }
+        const offer = await getBestOffer(params);
+
+        return offer;
+    }
+    
+    useEffect(() => {
+        getOffer().then((offer) => setBestOffer(offer));
```

Comment:
we'll handle catching in the next task

Reply:
100% agreed, not all calls are being handled properly